### ISSUE TITLE
Bug in DigraphDisjointUnion and ViewString for mutable digraphs

### ISSUE THREAD

**Issue by wilfwilson** [gap-packages/Digraphs#276]
Happening in `1.0.2`:
```
gap> D := EmptyDigraph(IsMutableDigraph, 3);
<mutable empty digraph with 3 vertices>
gap> DigraphDisjointUnion(D, CycleDigraph(3));;
gap> D;
<mutable acyclic digraph with 6 vertices, 3 edges>
gap> IsAcyclicDigraph(D);
true
gap> DigraphAllSimpleCircuits(D);
[ [ 4, 5, 6 ] ]
```
This is worrying for two reasons: how can a mutable digraph be storing an attribute (namely the property `IsAcyclicDigraph`), and how can it think that it is acyclic when it clearly isn't?

**Comment by james-d-mitchell** [comment]
Hmm excellent question

**Comment by wilfwilson** [comment]
I gets worse:
```
gap> List(D!.OutNeighbours, IsMutable);
[ true, true, true, false, false, false ]
```
These should all be mutable!

**Comment by james-d-mitchell** [comment]
Here's hoping this is a bug in `DigraphDIsjointUnion` and not something else :)

**Comment by wilfwilson** [comment]
Yep 🙂  Should be fine - I'll look into it shortly, just doing some experiments/'research' first (which is how I discovered this).

**Comment by wilfwilson** [comment]
Resolves in v1.0.3.
